src/rust/engine/process_executor/src/remote.rs

```diff
@@ -11,13 +13,49 @@ use super::{ExecuteProcessRequest, ExecuteProcessResult};
 /// Runs a command via a gRPC service implementing the Bazel Remote Execution API
 /// (https://docs.google.com/document/d/1AaGk7fOPByEvpAbqeXIyE8HX_A3_axxNnvroblTZ_6s/edit).
 ///
-pub fn run_command_remote(req: ExecuteProcessRequest) -> Result<ExecuteProcessResult, String> {
+pub fn run_command_remote(
+  addr: &str,
+  req: ExecuteProcessRequest,
+) -> Result<ExecuteProcessResult, String> {
+  let execute_request = make_execute_request(&req)?;
+
+  let env = Arc::new(grpcio::Environment::new(1));
```

Comment:
Because this reference does not escape the thread, I believe it can be an `Rc` rather than an `Arc`. Compilation will fail if something that needs to be thread safe is not, but it won't fail for the opposite case, afaik.

Reply:
Not a blocker though: will merge as is.

Reply:
The type signature of the thing it's passed to requires an Arc.